paasta_tools/setup_marathon_job.py

```diff
@@ -494,15 +518,29 @@ def log_deploy_error(errormsg: str, level: str='event') -> None:
 
     system_paasta_config = load_system_paasta_config()
     cluster = system_paasta_config.get_cluster()
-    existing_apps = marathon_tools.get_matching_apps(service, instance, marathon_apps)
-    new_app_list = [a for a in existing_apps if a.id == '/%s' % config['id']]
-    other_apps = [a for a in existing_apps if a.id != '/%s' % config['id']]
+    existing_apps_with_clients = marathon_tools.get_matching_apps_with_clients(
+        service,
+        instance,
+        marathon_apps_with_clients,
+    )
+
+    new_client = clients.get_current_client_for_service(job_config)
+
+    new_apps_with_clients_list: List[Tuple[MarathonApp, MarathonClient]] = []
```

Comment:
Technically speaking, an app doesn't bounce -- it gets created, scaled up, scaled down, stopped. A service/instance bounces.

Maybe think of it as new == desired, old == undesired. Most of the time new/old is correct, like when bouncing because a new version was marked for deployment, but sometimes new isn't accurate, like when rolling back.

Reply:
Agreed, naming is hard! I like desired/undesired. If you wanted to rename them to that I think it'll be easier to understand for the next person reading this section.